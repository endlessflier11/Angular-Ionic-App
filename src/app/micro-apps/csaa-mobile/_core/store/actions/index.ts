export * from './config.actions';
export * from './metadata.actions';
export * from './service-locator.actions';
export * from './auth.actions';
export * from './fetch.actions';
export * from './policy.actions';
export * from './customer.actions';
export * from './payment.actions';
export * from './claim.actions';
export * from './user-setting.actions';
