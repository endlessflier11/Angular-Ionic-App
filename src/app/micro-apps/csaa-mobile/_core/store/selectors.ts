import { FetchStatus } from '../interfaces';
import {
  AuthStateModel,
  CONFIG_STATE_TOKEN,
  ConfigStateModel,
  CSAA_APP_STATE_TOKEN,
  CSAA_AUTH_STATE_TOKEN,
  CUSTOMER_STATE_TOKEN,
  CustomerStateModel,
} from './states/state.interfaces';

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace ConfigSelector {
  export const state = (appState: { csaa_app: { csaa_config: ConfigStateModel } }) =>
    appState[CSAA_APP_STATE_TOKEN][CONFIG_STATE_TOKEN];

  export const configAndEndpointsLoaded = (appState: {
    csaa_app: { csaa_config: ConfigStateModel };
  }): boolean => {
    const configState = ConfigSelector.state(appState);
    return configState?.configLoaded && configState?.endpointsLoadState === FetchStatus.SUCCESS;
  };

  export const storeInitialized = (appState: {
    csaa_app: { csaa_config: ConfigStateModel };
  }): boolean => {
    const configState = ConfigSelector.state(appState);
    return configState?.storeInitialized;
  };
}

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace AuthSelector {
  export const state = (appState: { csaa_app: { csaa_auth: AuthStateModel } }) =>
    appState[CSAA_APP_STATE_TOKEN][CSAA_AUTH_STATE_TOKEN];
}

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace CustomerSelector {
  export const state = (appState: { csaa_app: { csaa_customer: CustomerStateModel } }) =>
    appState[CSAA_APP_STATE_TOKEN][CUSTOMER_STATE_TOKEN];
}
