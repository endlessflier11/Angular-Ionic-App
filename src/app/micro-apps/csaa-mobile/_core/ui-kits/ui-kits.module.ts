import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { ThemeIconComponent } from './theme-icon/theme-icon.component';
import { WorkingHoursComponent } from './working-hours/working-hours.component';
import { MyPolicyCardComponent } from './my-policy-card/my-policy-card.component';
import { SkeletonCardComponent } from './skeleton-card/skeleton-card.component';
import { BackButtonComponent } from './back-button/back-button.component';
import { PhonePipe } from './pipes/phone-number.pipe';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card/csaa-auto-id-card.component';
import { CsaaLoadingComponent } from './csaa-loading/csaa-loading.component';
import { ContentRendererComponent } from './content-renderer/content-renderer.component';
import { NationalCustomAlertModalComponent } from './national-custom-alert/national-custom-alert-modal';
import { SafeDataPipe } from './pipes/safe-data/safe-data.pipe';
import { DocumentActionModalComponent } from './document-action-modal/document-action-modal.component';
import { CsaaPolicySubtitlesComponent } from './csaa-policy-subtitles/csaa-policy-subtitles.component';
import { PaperlessTermsComponent } from './paperless-terms/paperless-terms.component';
import { PaperlessTermsModalComponent } from './paperless-terms-modal/paperless-terms-modal.component';
import { AppleWalletButtonComponent } from './apple-wallet-button/apple-wallet-button.component';

const UI_COMPONENTS = [
  ThemeIconComponent,
  WorkingHoursComponent,
  MyPolicyCardComponent,
  SkeletonCardComponent,
  BackButtonComponent,
  CsaaAutoIdCardComponent,
  CsaaLoadingComponent,
  ContentRendererComponent,
  NationalCustomAlertModalComponent,
  DocumentActionModalComponent,
  CsaaPolicySubtitlesComponent,
  PaperlessTermsComponent,
  PaperlessTermsModalComponent,
  AppleWalletButtonComponent,
];

const UI_PIPES = [PhonePipe, SafeDataPipe];

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, IonicModule],
  declarations: [...UI_COMPONENTS, ...UI_PIPES],
  exports: [...UI_COMPONENTS, ...UI_PIPES],
  providers: [DatePipe],
})
export class UiKitsModule {}
