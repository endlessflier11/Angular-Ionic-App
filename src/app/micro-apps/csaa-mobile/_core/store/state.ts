import { Injectable } from '@angular/core';
import { State } from '@ngxs/store';
import { ConfigState } from './states/config.state';
import { MetadataState } from './states/metadata.state';
import { CsaaAuthState } from './states/auth.state';
import { ContactInfoState } from './states/contact-info.state';
import { FetchState } from './states/fetch.state';
import { CUSTOMER_CHILD_STATES, CustomerState } from './states/customer.state';
import { CSAA_APP_STATE_TOKEN } from './states/state.interfaces';
import { UserSettingsState } from './states/user-setting.state';

/*
 * - __csaa_app
 * ├── config
 * ├── user_setting
 * ├── metadata
 * ├── contact // TODO: consume state withing the app
 * ├── fetch
 * ├── auth
 * └── customer
 *     ├── policies
 *     ├── payments
 *     └── claims
 * */
const CHILD_STATES = [
  ConfigState,
  MetadataState,
  ContactInfoState,
  CsaaAuthState,
  FetchState,
  CustomerState,
  UserSettingsState,
];

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface CsaaAppStateModel {}

@State<CsaaAppStateModel>({
  name: CSAA_APP_STATE_TOKEN,
  children: CHILD_STATES,
})
@Injectable()
export class CsaaAppState {}

export const CSAA_STORES = [CsaaAppState, ...CHILD_STATES, ...CUSTOMER_CHILD_STATES];
