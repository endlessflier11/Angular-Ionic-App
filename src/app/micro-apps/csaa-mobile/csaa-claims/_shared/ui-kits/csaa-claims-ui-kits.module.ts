import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { AgentCardComponent } from './agent-card/agent-card.component';
import { ClaimsHistoryCardComponent } from './claims-history-card/claims-history-card.component';
import { ClaimsInfoCardComponent } from './claims-info-card/claims-info-card.component';
import { WhatToDoCardComponent } from './what-to-do-card/what-to-do-card.component';

const SHARED_COMPONENTS = [
  AgentCardComponent,
  ClaimsHistoryCardComponent,
  ClaimsInfoCardComponent,
  WhatToDoCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaClaimsUiKitsModule {}
