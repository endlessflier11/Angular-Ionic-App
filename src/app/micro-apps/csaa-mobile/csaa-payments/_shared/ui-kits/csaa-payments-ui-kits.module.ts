import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
