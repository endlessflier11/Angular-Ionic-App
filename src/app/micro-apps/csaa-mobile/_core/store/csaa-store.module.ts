import { NgModule } from '@angular/core';
import { environment } from 'src/environments/environment';
import { NgxsModule } from '@ngxs/store';
import { CSAA_STORES } from './state';

@NgModule({
  imports: [
    NgxsModule.forRoot(CSAA_STORES, {
      developmentMode: !environment.production,
    }),
    NgxsModule.forFeature(CSAA_STORES),
  ],
})
export class CsaaStoreModule {}
