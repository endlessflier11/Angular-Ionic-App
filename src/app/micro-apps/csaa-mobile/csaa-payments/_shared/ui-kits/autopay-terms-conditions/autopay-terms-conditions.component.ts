import { Component } from '@angular/core';
import { map } from 'rxjs/operators';
import { ModalController } from '@ionic/angular';
import { LegalService } from '../../../../_core/services/legal.service';
import { noop, Observable } from 'rxjs';
import { Store } from '@ngxs/store';
import { ConfigState } from '../../../../_core/store/states/config.state';

@Component({
  selector: 'csaa-autopay-terms-conditions',
  templateUrl: './autopay-terms-conditions.component.html',
})
export class AutopayTermsConditionsComponent {
  currentTheme: string;
  source$: Observable<string>;

  constructor(
    private readonly store: Store,
    private legalService: LegalService,
    private modalCtrl: ModalController
  ) {
    this.currentTheme = this.store.selectSnapshot(ConfigState.theme);
    this.source$ = this.legalService.loadAutopayTerms().pipe(map(({ body }) => body));
  }

  dismissModal() {
    this.modalCtrl.dismiss().then(noop);
  }
}
