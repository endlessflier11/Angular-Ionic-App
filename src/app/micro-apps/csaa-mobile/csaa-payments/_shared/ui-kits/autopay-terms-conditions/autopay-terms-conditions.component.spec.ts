import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By, StoreTestBuilder, PageTestingModule } from '@app/testing';
import { IonicModule } from '@ionic/angular';
import { of } from 'rxjs';
import { LegalService } from '../../../../_core/services/legal.service';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';

import { AutopayTermsConditionsComponent } from './autopay-terms-conditions.component';
import { Store } from '@ngxs/store';
import { CsaaCoreModule } from '../../../../csaa-core/csaa-core.module';
try {
  describe('AutopayTermsConditionsComponent', () => {
    let component: AutopayTermsConditionsComponent;
    let fixture: ComponentFixture<AutopayTermsConditionsComponent>;

    beforeEach(async(() => {
      try {
        TestBed.configureTestingModule({
          declarations: [AutopayTermsConditionsComponent],
          providers: [
            {
              provide: LegalService,
              useValue: {
                loadAutopayTerms: jest
                  .fn()
                  .mockReturnValue(of({ body: '<h1>Terms and conditions</h1> Lore ipsum ...' })),
              },
            },
          ],
          imports: [
            PageTestingModule.withConfig({
              providesConfig: true,
              providesStorage: true,
              providesRouter: true,
              providesModal: true,
            }),
            IonicModule,
            UiKitsModule,
            CsaaCoreModule,
          ],
        }).compileComponents();
        const store = TestBed.inject(Store);
        StoreTestBuilder.withDefaultMocks().resetStateOf(store);

        fixture = TestBed.createComponent(AutopayTermsConditionsComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      } catch (error) {
        console.error(error);
        fail(error.message);
      }
    }));

    it('should create', () => {
      expect(component).toBeTruthy();
    });
    it('should match snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });
    it('should navigate to the back page when back button is clicked', () => {
      const backButton = fixture.debugElement.query(By.css('csaa-back-button'));
      expect(backButton).toBeTruthy();
    });
  });
} catch (error) {
  console.error(error);
  throw error;
}
