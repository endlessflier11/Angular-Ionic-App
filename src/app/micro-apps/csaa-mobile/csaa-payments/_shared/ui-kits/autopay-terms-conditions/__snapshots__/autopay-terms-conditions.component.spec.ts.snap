// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutopayTermsConditionsComponent should match snapshot 1`] = `
<csaa-autopay-terms-conditions
  __ngContext__={[Function LRootView]}
  currentTheme=""
  legalService={[Function Object]}
  modalCtrl={[Function AlertControllerMock]}
  source$={[Function Observable]}
  store={[Function Store]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar
      no-border=""
    >
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-title>
        Terms & Conditions
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <csaa-content-renderer>
      <div
        class="legal"
      >
        <h1>
          Terms and conditions
        </h1>
         Lore ipsum ...
      </div>
    </csaa-content-renderer>
  </ion-content>
</csaa-autopay-terms-conditions>
`;
